tests/Sharpliner.Serialization.Tests/MockPipeline.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Sharpliner.Model.AzureDevOps;
+
+namespace Sharpliner.Serialization.Tests
+{
+    internal class MockPipeline : AzureDevOpsPipelineDefinition
+    {
+        public override string TargetFile => "azure-pipelines.yml";
+
+        public override TargetPathType TargetPathType => TargetPathType.RelativeToGitRoot;
+
+        public override Pipeline Pipeline => new()
+        {
+            Name = "$(Date:yyyMMdd).$(Rev:rr)",
+
+            Variables =
+            {
+                new Variable("Configuration", "Release"), // We can create the objects and then resue them for definition too
+                Variable("Configuration", "Release"),     // Or we have this more YAML-like definition
+                Group("PR keyvault variables"),
+
+                If.Equal("variables['Build.Reason']", "PullRequest")
+                    .Variable("TargetBranch", "$(System.PullRequest.SourceBranch)")
+                    .Variable("IsPr", true),
```

Comment:
Static analysis fo the variable names to be unique is going to be interesting, but is doable with this syntax.

Reply:
I went through some of our YAML files and we always group more definitions under one conditions so I wanted to have that here